### ISSUE TITLE
Argument count is off for functions/methods when using "for"-loops

### ISSUE THREAD

**Issue by noctux** [castwide/solargraph#349]
Hello,

Thank you for this nice languageserver!

When analyzing the following code:

```ruby
# frozen_string_literal: true

# Print stuff reversed
def print_reverse(items)
  for i in (items.length - 1).downto(0) do
    puts items[i]
  end
end

def do_something()
  things=["a", "b", "c"]
  print_reverse(things)
end

do_something
```

...solargraph somehow gets confused about the arity of `print_reverse`, and throws an error `Not enough arguments to #print_reverse` (please ignore the typo in the documentation comment...):

![2020-08-06-135753_849x338_scrot](https://user-images.githubusercontent.com/291559/89529891-d04ab780-d7ed-11ea-8237-f8537faa25c6.png)

The additional comma in the signature indicates that solargraph believes that there is indeed an invisible second argument to `print_reverse`. I'm not that familiar with ruby so there might actually be, however the code above works fine for me, and prints `c\nb\na\n` as expected, so I guess it's acceptable ruby?

(This is lsp-mode in emacs, I don't know how to trigger the behaviour at the cli alone, sorry)

The issue happened to me for various functions, all using `for`-loops in a bigger project.

I'm not 100% sure whether this is a valid bug or what information you need in detail, so please do ping me if you need more information.

Thank you for taking interest,
Simon

Additional Information:
- `ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux]`
- solargraph Version `0.39.13`
- emacs version `26.3`, lsp-mode `20200804.1536`
- [Gemfile.lock.txt](https://github.com/castwide/solargraph/files/5035200/Gemfile.lock.txt) for the dependencies

Edit: Forgot to add my `.solargraph.yml`. Basically the default one, but with `typecheck:typed`:

```YML
---
include:
- "**/*.rb"
exclude:
- spec/**/*
- test/**/*
- ".vendor/**/*"
- ".bundle/**/*"
require: []
domains: []
reporters:
- rubocop
- typecheck:typed
- require_not_found
require_paths: []
plugins: []
max_files: 5000
```

**Comment by castwide** [comment]
Confirmed. I'll work on a fix for the next patch release.

**Comment by noctux** [comment]
Perfect, thanks!

I believe it is not super urgent (rephrasing it as an `.each` loop works just fine), but of course it would be nice if it worked one day :)

**Comment by castwide** [comment]
Released in 0.39.14.
